components/script/dom/htmlbodyelement.rs

```diff
@@ -94,16 +87,27 @@ impl HTMLBodyElementMethods for HTMLBodyElement {
     }
 }
 
+pub trait HTMLBodyElementLayoutHelpers {
+    fn get_background_color(&self) -> Option<RGBA>;
+    fn get_color(&self) -> Option<RGBA>;
+    fn get_background(&self) -> Option<Url>;
+}
 
-impl HTMLBodyElement {
-    pub fn get_background_color(&self) -> Option<RGBA> {
-        self.background_color.get()
+impl HTMLBodyElementLayoutHelpers for LayoutJS<HTMLBodyElement> {
+    #[allow(unsafe_code)]
+    fn get_background_color(&self) -> Option<RGBA> {
+        unsafe {
+            (&*self.upcast::<Element>().unsafe_get())
+                .get_attr_for_layout(&ns!(""), &atom!("bgcolor"))
```

Comment:
Should this be `bgColor`?

I actually don't know, just pointing out one of these might be capitalized